src/components/FileUpload.tsx

```diff
@@ -96,17 +189,19 @@ export default function FileUpload() {
                                 })
                         )
                     );
-                    console.table({ dsChanged, datasets });
                     if (dsChanged) {
-                        console.log(datasets);
                         setDatasets({ ...datasets });
                         // @ts-ignore
-                        const toolbox: Blockly.Toolbox = Blockly.getMainWorkspace().toolbox_;
-                        toolbox.setSelectedItem(toolbox.contents_
-                            .filter(content => content
-                                // @ts-ignore
-                                .name_
-                                === "DataFrames")[0]);
+                        const toolbox: Blockly.Toolbox = (
```

Comment:
In the toolbox we have a thing that automatically shows a block for each uploaded dataset. This code is responsible for opening that tab to show the user where they can access their new dataset. We need to rethink how this will work. I think that it should work like this:

- The datasets tab shows a list of all datasets the user has ever uploaded
- Next to each dataset is a button to include the dataset in the project
- When it is included it is added to the `datasets` array, so we don't need to change much other code and the smart toolbox thing still works

This also solves the problem of how permissions for datasets work; they should have the same permissions as the project they're used in. So, the API would need to check when a user tries to see a dataset if the dataset is **either** owned by the user or is part of a public project.